attachment_synchronize_pattimpex/models/attachment_synchronize_task.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, fields, models
+from odoo.osv import expression
+from odoo.tools import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+class AttachmentSynchronizeTask(models.Model):
+    _inherit = "attachment.synchronize.task"
+
+    method_type = fields.Selection(
+        selection_add=[
+            ("import_pattern", "Import Task using patterns"),
+            ("export_pattern", "Export task using patterns"),
+        ],
+        required=True,
+    )
+    domain_pattimpex_export = fields.Char(
+        string="Domain for filtering records to export", default="[]"
+    )
+    export_id = fields.Many2one("ir.exports", string="Import/Export pattern")
+
```

Comment:
I don't think we want to add "methode types" here.
These synchronize tasks are juste made to store the configuration to export a file to an external system (sftp/ftp, etc...) or to import a task for an external system. 
When you want to import a file from a sftp, you define the methode_type 'import' and then you fill the `file_type` with how you want to process the file once it is imported. here we should do
`file_type = field.Selection(selection_add=[('import_pattern', "Import pattern")])`

For the export part, it is questionable. I find it a bit confusing to add this `export_pattern` `methode_type`
For me its goal is to store the configuration of the external system where we want to send some files (server, port, etc...) and I am not sure it should be the place to store the fact that we want to automate some exports.
Because, if we do it this way, with other module we could have multiple 'methode_type' export_***
But at the end, the system do just exactly the same, it will export a file to an external system. That is why I find it confuse to have multiple "export_****" `methode_type`
But I understand it is convenient to put this here...